--- /usr/share/voicecall-ui-jolla/qml/pages/calllog/CallLogItem.qml.bak 2014-07-18 11:12:42.059709577 +0200
+++ /usr/share/voicecall-ui-jolla/qml/pages/calllog/CallLogItem.qml     2014-07-18 11:17:14.494139675 +0200
@@ -23,6 +23,7 @@
     property bool showNumberDetail
     property string numberDetail: showNumberDetail ? getNumberDetail() : ""
     property string connectionDetail: connection.length && connection !== "Mobile" ? connection : ""
+    property string phoneNumber: person ? CallLog.formatNumber(remoteUid) : ""
     property real nameRightPadding

     width: parent ? parent.width : 0
@@ -110,10 +111,10 @@
     Label {
         anchors {
             top: description.bottom
-            right: description.right
+            left: description.left
         }
         font.pixelSize: Theme.fontSizeExtraSmall
-        text: numberDetail + (numberDetail.length && connectionDetail.length ? " | " : "") + connectionDetail
+        text: phoneNumber + (phoneNumber.length && numberDetail.length ? " | " : "") + numberDetail + (numberDetail.length && connectionDetail.length ? " | " : "") + connectionDetail
         color: highlighted ? Theme.secondaryHighlightColor : secondaryColor
     }
