diff -rupN original/usr/share/voicecall-ui-jolla/qml/pages/calllog/CallLogItem.qml patched/usr/share/voicecall-ui-jolla/qml/pages/calllog/CallLogItem.qml
--- original/usr/share/voicecall-ui-jolla/qml/pages/calllog/CallLogItem.qml	2014-06-10 10:03:30.000000000 +0100
+++ patched/usr/share/voicecall-ui-jolla/qml/pages/calllog/CallLogItem.qml	2014-06-10 10:57:58.985277433 +0100
@@ -61,6 +61,11 @@ ListItem {
         height: firstNameText.height
         baselineOffset: firstNameText.baselineOffset
         Row {
+            id: nameRow
+            spacing: Theme.paddingSmall
+            width: parent.width - (timeStampLabel.visible ? timeStampLabel.width + Theme.paddingMedium : 0)
+
+            Row{
             spacing: Theme.paddingSmall
             width: parent.width - (timeStampLabel.visible ? timeStampLabel.width + Theme.paddingMedium : 0)
 
@@ -84,6 +89,23 @@ ListItem {
                 width: Math.min(implicitWidth, parent.width - firstNameText.width)
                 visible: width > 0
             }
+            }
+            Row {
+                spacing: Theme.paddingSmall
+                width: parent.width - (timeStampLabel.visible ? timeStampLabel.width + Theme.paddingMedium : 0)
+                anchors.top: nameRow.bottom
+
+                Label {
+                    id: phoneNumber
+                    anchors.left: parent.left
+                    opacity: privateNumber ? 0.5 : 1.0
+                    text: main.formatNumber(remoteUid)
+                    color: highlighted ? Theme.secondaryHighlightColor : secondaryColor
+                    font.pixelSize: Theme.fontSizeExtraSmall
+                    truncationMode: TruncationMode.Fade
+                    width: Math.min(implicitWidth, parent.width)
+                }
+            }
         }
         Label {
             id: timeStampLabel
