--- /usr/lib/qt5/qml/com/jolla/mediaplayer/AudioPlayer.qml	2000-01-01 01:00:00.000000000 +0100
+++ /usr/lib/qt5/qml/com/jolla/mediaplayer/AudioPlayer.qml	2015-06-09 07:46:21.975996782 +0200
@@ -127,6 +127,10 @@
         audio.model.appendUrl(url)
         playIndex(0)
     }
+    function toggleOrNext(){
+	toggleTimer.start()
+	toggle()
+    }
 
     function toggle() {
         if (playing) {
@@ -194,7 +198,8 @@
     MediaKey { enabled: player._grabKeys; key: Qt.Key_MediaStop; onReleased: audio.stop() }
     MediaKey { enabled: player._grabKeys; key: Qt.Key_MediaNext; onReleased: audio.playNext() }
     MediaKey { enabled: player._grabKeys; key: Qt.Key_MediaPrevious; onReleased: audio.playPrevious() }
-    MediaKey { enabled: player._grabKeys; key: Qt.Key_ToggleCallHangup; onReleased: player.toggle() }
+    MediaKey { enabled: player._grabKeys; key: Qt.Key_ToggleCallHangup; onPressed: player.toggleOrNext(); onReleased: toggleTimer.stop() }
+    Timer { id: toggleTimer; interval: 400; onTriggered: audio.playNext()}
 
     MediaKey {
         id: forwardKey
